api/app/services/faq_service.py

```
@@ -685,47 +796,100 @@ def extract_faqs_with_openai(self, conversations: List[Dict]) -> List[Dict]:
-                    'question': faq.get('question', ''),
-                    'answer': faq.get('answer', ''),
-                    'category': faq.get('category', 'General'),
-                    'source': faq.get('source', 'Bisq Support Chat')
-                }
-                f.write(json.dumps(normalized_faq, ensure_ascii=False) + '\n')
-        logger.info(f"Saved {len(faqs)} FAQ entries to {self.faq_file_path}")
+        # Now save the merged and deduplicated list
+        try:
+            with portalocker.Lock(str(self.faq_file_path), mode='w',
+                                  timeout=10) as f_lock:
+                with open(self.faq_file_path, "w", encoding="utf-8") as f:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)